src/App.jsx

```diff
@@ -14,33 +15,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortType, setSortType] = useState(null);
+  const [checkRevers, setCheckReverse] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSort = type => {
+    setSortType(type);
+  };
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortType === 'alphabetical') {
+    visibleGoods.sort();
+  } else if (sortType === 'length') {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (checkRevers) {
+    visibleGoods.reverse();
+  }
+
+  const reversedGoods = () => {
+    setCheckReverse(!checkRevers);
+  };
+
+  const resetState = () => {
+    setSortType(null);
+    setCheckReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabetical' ? '' : 'is-light'}`}
+          onClick={() => handleSort('alphabetical')}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${sortType === 'length' ? '' : 'is-light'}`}
+          onClick={() => handleSort('length')}
```

Comment:
No need in additional function here
we can use directly
```suggestion
          onClick={() => setSortType('length')}
```